Tests/UnitTests/Core/TrackFitting/Gx2fTests.cpp

```diff
@@ -301,27 +321,35 @@ BOOST_AUTO_TEST_CASE(Fit5Iterations) {
   extensions.surfaceAccessor
       .connect<&TestSourceLink::SurfaceAccessor::operator()>(&surfaceAccessor);
 
-  MagneticFieldContext mfContext;
-  CalibrationContext calContext;
-
   const Experimental::Gx2FitterOptions gx2fOptions(
-      tgContext, mfContext, calContext, extensions, PropagatorPlainOptions(),
-      rSurface, false, false, FreeToBoundCorrection(false), 5, true, 0);
+      geoCtx, magCtx, calCtx, extensions, PropagatorPlainOptions(), rSurface,
+      false, false, FreeToBoundCorrection(false), 5, true, 0);
 
   Acts::TrackContainer tracks{Acts::VectorTrackContainer{},
                               Acts::VectorMultiTrajectory{}};
 
-  // Fit the track
-  auto res = fitter.fit(sourceLinks.begin(), sourceLinks.end(),
-                        startParametersFit, gx2fOptions, tracks);
+  ACTS_DEBUG("Fit the track");
+  ACTS_VERBOSE("startParameter unsmeared:\n" << parametersMeasurements);
+  ACTS_VERBOSE("startParameter fit:\n" << startParametersFit);
+  const auto res = fitter.fit(sourceLinks.begin(), sourceLinks.end(),
+                              startParametersFit, gx2fOptions, tracks);
 
   BOOST_REQUIRE(res.ok());
 
-  auto& track = *res;
+  const auto& track = *res;
+
   BOOST_CHECK_EQUAL(track.tipIndex(), nSurfaces - 1);
   BOOST_CHECK(track.hasReferenceSurface());
-  BOOST_CHECK_EQUAL(track.nMeasurements(), nSurfaces);
+
+  // Track quantities
+  CHECK_CLOSE_ABS(track.chi2(), 8., 2.);
```

Comment:
Here we have the same problem as with the final parameters. Since on different OS/architecture/compiler/??? (we haven't figured it out in detail) the RNG gives a different smearing to the measurement, our fitted trajectory differs a bit. There we decided to be "close" to the true parameters. For the chi2 the resulting effect seems to be a lot stronger, but when looking at the chi2-value-space, it isn't too bad. In the beginning of the fit we have a huge chi2, could be even 1e10. So converging to `6` or `10` is both good enough.

I thought about setting it to `BOOST_CHECK_LT(track.chi2(), 10);` but then we would:
- need an extra test to check `chi2 != 0`
- won't detect, when the chi2 somehow changes to a smaller value